docs/modules/ROOT/pages/explanations/jsonnet.adoc

```diff
@@ -0,0 +1,46 @@
+= Jsonnet Best Practices
+
+== Merging External Config Data
+* Deep merge helper
+* Merge from hiera
+* Expose defaults in hiera
+* Dict with key=name (too keep them mergeable)
+* Provide "generic" defaults and merge with context specifics (i.e. cloud/region see openshift4-nodes)
+* Allow Null (to enable empty)
+* For non-mergeable data (i.e. strings), provide defaults + extras
```

Comment:
Probably should add something along the lines of "try to avoid exposing non-mergeable data directly in the hierarchy, consider implementing rendering structured data into string format when the component is compiled"